.gitlab-ci.yml

```
@@ -728,14 +729,23 @@ publish-rustdoc:
   stage:                           publish
   <<:                              *kubernetes-env
   <<:                              *vault-secrets
-  image:                           paritytech/tools:latest
+  image:                           node:16
   variables:
     GIT_DEPTH:                     100
+    # --- Following variables are for rustdocs deployment ---
+    # Space separated values of branches and tags to generate rustdocs
+    RUSTDOCS_DEPLOY_REFS:          "master monthly-2021-09+1 monthly-2021-08 v3.0.0"
+    # Location of the docs index template
+    INDEX_TPL:                     ".maintain/docs-index-tpl.ejs"
+    # Where the `/latest` symbolic link links to. One of the $RUSTDOCS_DEPLOY_REFS value.
+    LATEST:                        "monthly-2021-09+1"
   rules:
     - if: $CI_PIPELINE_SOURCE == "pipeline"
       when: never
     - if: $CI_PIPELINE_SOURCE == "web" && $CI_COMMIT_REF_NAME == "master"
     - if: $CI_COMMIT_REF_NAME == "master"
+    - if: $CI_COMMIT_REF_NAME =~ /^monthly-20[0-9]{2}-[0-9]{2}.*$/  # to support: monthly-2021-09+1
+    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/
```

```suggestion
    - if: $CI_COMMIT_REF_NAME =~ /^v[0-9]+\.[0-9]+.*$/              # i.e. v1.0, v2.1rc1
```
let's comment this too